agent/base_agent.py

```diff
@@ -116,7 +116,10 @@ def _format_bash_execution_result(
       previous_prompt: Optional[Prompt] = None) -> str:
     """Formats a prompt based on bash execution result."""
     if previous_prompt:
-      previous_prompt_text = previous_prompt.get()
+      if isinstance(previous_prompt, OpenAIPrompt):
+        previous_prompt_text = previous_prompt.gettext()
+      else:
+        previous_prompt_text = previous_prompt.get()
```

Comment:
Thanks for noticing and fixing this, @arthurscchan.
Would you think me could add a new `gettext()` method for `TextPrompt` (and `Prompt`) too? The new method returns the same value as `get()` in `TextPrompt`.
This allows us to call `previous_prompt.gettext()` transparently for both `TextPrompt` and `OpenAIPrompt`.

The main intention is to avoid special cases (i.e., `if isinstance...`) because over time more and more special cases will become harder to maintain.

Reply:
Thanks @arthurscchan  : )